gradio/interface.py

```diff
@@ -654,65 +651,61 @@ def attach_submit_events(self, submit_btn: Button | None, stop_btn: Button | Non
-                        inputs=None,
-                        outputs=[submit_btn, stop_btn],
-                        queue=False,
-                    ).then(
-                        self.fn,
-                        self.input_components,
-                        self.output_components,
-                        api_name=self.api_name if i == 0 else None,
-                        scroll_to_output=True,
-                        preprocess=not (self.api_mode),
-                        postprocess=not (self.api_mode),
-                        batch=self.batch,
-                        max_batch_size=self.max_batch_size,
-                    )
-                    predict_events.append(predict_event)
-
-                    predict_event.then(
-                        cleanup,
-                        inputs=None,
-                        outputs=extra_output,  # type: ignore
-                        queue=False,
-                    )
+                predict_event = on(
+                    triggers,
+                    lambda: (
+                        submit_btn.update(visible=False),
+                        stop_btn.update(visible=True),
+                    ),
+                    inputs=None,
+                    outputs=[submit_btn, stop_btn],
+                    queue=False,
+                ).then(
+                    self.fn,
+                    self.input_components,
+                    self.output_components,
+                    api_name=self.api_name,
+                    scroll_to_output=True,
+                    preprocess=not (self.api_mode),
+                    postprocess=not (self.api_mode),
+                    batch=self.batch,
+                    max_batch_size=self.max_batch_size,
+                )
+
+                predict_event.then(
+                    cleanup,
+                    inputs=None,
+                    outputs=extra_output,  # type: ignore
+                    queue=False,
+                )
 
                 stop_btn.click(
                     cleanup,
                     inputs=None,
                     outputs=[submit_btn, stop_btn],
-                    cancels=predict_events,
+                    cancels=predict_event,
                     queue=False,
                 )
             else:
-                for i, trigger in enumerate(triggers):
-                    predict_events.append(
-                        trigger(
-                            fn,
-                            self.input_components,
-                            self.output_components,
-                            api_name=self.api_name if i == 0 else None,
-                            scroll_to_output=True,
-                            preprocess=not (self.api_mode),
-                            postprocess=not (self.api_mode),
-                            batch=self.batch,
-                            max_batch_size=self.max_batch_size,
-                        )
-                    )
+                on(
```

Comment:
But the original behavior is a bug right? We always block identical events from triggering each other in 3.x (we plan to change this a bit in 4.0 to ensure that the final event is sent through) -- so this would actually be making it more consistent with the way events are handled.

Reply:
yeah it's actually a bug that you can trigger another submit through a different trigger. I see this as a bugfix.